src/expr/proof_node_manager.cpp

```diff
@@ -134,34 +135,60 @@ std::shared_ptr<ProofNode> ProofNodeManager::mkScope(
     //  (= false (= t1 t2))
     bool pol, symm;
     Node aPred = CDProof::getPredicateFact(a, pol, symm);
-    if (!aPred.isNull() && ac.count(aPred))
+    Trace("pnm-scope") << "  - try predicate fact " << aPred << "\n";
+    if (!aPred.isNull())
     {
-      Trace("pnm-scope") << "- introduce explicit predicate " << aPred
-                         << " via " << (symm ? "[reoriented] " : "") << a
-                         << " for " << fa.second.size() << " proof nodes"
-                         << std::endl;
-      std::shared_ptr<ProofNode> pfaa = mkAssume(aPred);
-      ;
-      // maybe reorient equality, in which case update the pointer to the
-      // justification of the assumption
-      if (symm)
+      // Whether the predicate also needs to be flipped to a get a bound
+      // assumption.
+      bool flipPred = false;
+      // Whether we've found a path to a bound assumption.
+      bool foundMatch = false;
+      Node symPred;
+      if (ac.count(aPred))
       {
-        pfaa = mkNode(pol ? PfRule::TRUE_INTRO : PfRule::FALSE_INTRO,
-                      pfaa,
-                      {},
-                      a[1].eqNode(a[0]));
+        foundMatch = true;
       }
-      PfRule updateRule =
-          symm ? PfRule::SYMM
-               : (pol ? PfRule::TRUE_INTRO : PfRule::FALSE_INTRO);
-      for (ProofNode* pfs : fa.second)
+      else
       {
-        Assert(pfs->getResult() == a);
-        updateNode(pfs, updateRule, {pfaa}, {});
+        symPred = CDProof::getSymmFact(aPred);
+        if (!symPred.isNull() && ac.count(symPred))
+        {
+          foundMatch = true;
+          flipPred = true;
+        }
+      }
+      if (foundMatch)
+      {
+        Trace("pnm-scope") << "- introduce explicit predicate " << aPred
+                           << " via " << (symm ? "[reoriented] " : "") << a
+                           << " for " << fa.second.size() << " proof nodes"
+                           << std::endl;
+        std::shared_ptr<ProofNode> pfaa = mkAssume(flipPred ? symPred : aPred);
+        // Potentially apply symmetry to the real assumption
+        if (flipPred) {
+          pfaa = mkNode(PfRule::SYMM, pfaa, {}, aPred);
+        }
+        // maybe reorient equality, in which case update the pointer to the
+        // justification of the assumption
+        if (symm)
+        {
+          pfaa = mkNode(pol ? PfRule::TRUE_INTRO : PfRule::FALSE_INTRO,
+                        pfaa,
+                        {},
+                        a[1].eqNode(a[0]));
+        }
+        PfRule updateRule =
+            symm ? PfRule::SYMM
```

Comment:
Now that we have two kind of symmetries for this case it'd be good to rename this variable. Following the new one you created we could call this `flipEqualityPredIntro`.

Reply:
Done in 4fe9d7bee